CHANGELOG.md

```diff
@@ -1,3 +1,300 @@
+
+##### Old:
+
+```hcl
+packet {
+  oidc {
+    client_id = "gangway"
+  }
+}
+```
+
+##### New:
+
+```hcl
+packet {
+  oidc {
+    client_id = "clusterauth"
+  }
+}
+```
+
+#### Cluster update steps
+
+Ensure, that your cluster is in a healthy state by running `lokoctl cluster apply` using `v0.4.1` version. As we currently do not support version-jumping, so if your cluster is older, then upgrade to `v0.4.1` and only then proceed with upgrades to `v0.5.0`.
+
+Due to [Terraform](https://github.com/kinvolk/lokomotive/pull/824) and [Kubernetes](https://github.com/kinvolk/lokomotive/pull/1030) updates to v0.13+ and v1.19.3 respectively, some manual steps need to be performed when updating. In your cluster configuration directory, follow these steps:
+
+1. Update local Terraform binary to version v0.13.X. You can follow [this guide](https://learn.hashicorp.com/tutorials/terraform/install-cli.) to do that.
+
+2. Starting from your cluster directory, export your platform name and assets directory name used in your platform configuration. It will be used in next steps:
+  ```sh
+  export PLATFORM="packet" && export ASSETS_DIR="assets"
+  ```
+
+3. Remove old asset files:
+  ```sh
+  rm -f $ASSETS_DIR/terraform-modules/$PLATFORM/flatcar-linux/kubernetes/require.tf \
+  $ASSETS_DIR/terraform-modules/$PLATFORM/flatcar-linux/kubernetes/workers/require.tf \
+  $ASSETS_DIR/terraform-modules/dns/route53/require.tf
+  ```
+
+4. Go to the `terraform` directory:
+  ```sh
+  cd $ASSETS_DIR/terraform
+  ```
+
+5. Replace the old providers:
+  ```sh
+  terraform state replace-provider -auto-approve registry.terraform.io/-/ct registry.terraform.io/poseidon/ct && \
+  terraform state replace-provider -auto-approve registry.terraform.io/-/template registry.terraform.io/hashicorp/template
+  ```
+
+6. Return to original directory and use kubeconfig generated by lokomotive:
+
+  ```sh
+  cd - && export KUBECONFIG=$ASSETS_DIR/cluster-assets/auth/kubeconfig
+  ```
+
+7. `FelixConfiguration` has been moved to calico charts. To avoid firewall blip, label and annotate it, so that it can be managed by Helm while updating:
+  ```sh
+  kubectl label FelixConfiguration default app.kubernetes.io/managed-by=Helm --overwrite=true && \
+  kubectl annotate FelixConfiguration default meta.helm.sh/release-name=calico --overwrite=true && \
+  kubectl annotate FelixConfiguration default meta.helm.sh/release-namespace=kube-system --overwrite=true
+  ```
+
+Finally, run the following:
+
+```sh
+lokoctl cluster apply --skip-components -v
+```
+
+**NOTE:** On clusters with a single controller node only, delete the old `kube-apiserver` ReplicaSet during cluster update:
+  ```sh
+  kubectl delete rs -n kube-system $(kubectl get rs -n kube-system -l k8s-app=kube-apiserver --no-headers=true --sort-by=metadata.creationTimestamp | tac | tail -n +2 | awk '{print $1}') || true
```

Comment:
Yes, it is expected that update process will get interrupted (as we remove old pod and we must wait for new one).

>How does this work? This will only apply when there are two apiserver replicasets, right? In that case maybe it makes sense to make a loop so the user can run it while the cluster is updating?

During the upgrade, the new one should get created, but it will never converge. At this point, removing old one allows it to converge.

If there is just one for whatever reason, this comment just do not anything, so it's safe. It's also idempotent, just in case.